src/utils/re_frame.cljs

```diff
@@ -76,6 +77,14 @@
     (swap! handler-nesting-level dec)
     res))
 
+(defn delay-render
+  [content]
+  (let [render? (reagent/atom false)]
+    (js/setTimeout #(reset! render? true) 0)
```

Comment:
Have you considered `setImmediate`?

Reply:
no why? setTimeout fits perfectly here

Reply:
Just because of reading.

We are setting a timeout as zero, so there is no "time" involved. `SetImmediate` sounds as what we want :thinking: 

Now I'm wondering if removing the timeout gives the same result:

```clojure
(let [render? (reagent/atom false)
      _       (reset! render? true)]
  (fn [content] 
    ,,,))
```

Since once an atom is changed, reagent perform the updates the next cycle, so it means the first time the component is rendered using `false` and the second time with `true` :thinking:

Reply:
probably, but for me `(js/setTimeout  0)` is known thing, so i'm sure it works how it should

Reply:
Update on this:
It won't work, the atom is always received in the render function as `false`.
I also tried using `setImmediate` (to trigger the atom `reset!` at the end of the JS cycle) and it works too!

One thing I've just noticed is that we are **not** receiving `content` in the render function, it means, if `content` changes, it won't trigger a re-render.

Is this by mistake or intended?
As @ilmotta said, since this is a general purpose component, and `content` can be anything, maybe we should receive `content` in the render fn

Reply:
in this case it's intentional,ok i think it would be better to move it to profile settings namespace, because there is confusion now, this was done only for this current case but for some reason i decided to put it into utils re-frame namespace